src/Simplify_Call.cpp

```diff
@@ -186,14 +186,23 @@ Expr Simplify::visit(const Call *op, ExprInfo *info) {
             return Call::make(op->type, result_op, {a, b}, Call::PureIntrinsic);
         }
     } else if (op->is_intrinsic(Call::bitwise_and)) {
-        Expr a = mutate(op->args[0], nullptr);
-        Expr b = mutate(op->args[1], nullptr);
+        ExprInfo a_info, b_info;
+        Expr a = mutate(op->args[0], &a_info);
+        Expr b = mutate(op->args[1], &b_info);
 
         Expr unbroadcast = lift_elementwise_broadcasts(op->type, op->name, {a, b}, op->call_type);
         if (unbroadcast.defined()) {
             return mutate(unbroadcast, info);
         }
 
+        if (info && (op->type.is_int() || op->type.is_uint())) {
+            auto bits_known = a_info.to_bits_known(op->type) & b_info.to_bits_known(op->type);
+            info->from_bits_known(bits_known, op->type);
+            if (bits_known.mask == (uint64_t)-1) {
```

Comment:
Add comment saying something like "This is a constant". Or make `ExprInfo::BitsKnown` have a helper function like "std::optional<uint64_t> as_constant()`.

Reply:
and same below with `bitwise_or`